community/kernel/src/test/java/org/neo4j/unsafe/impl/batchimport/input/csv/CsvInputTest.java

```diff
@@ -482,6 +488,156 @@ public void shouldParsePointPropertyValues() throws Exception
         }
     }
 
+    @Test
+    public void shouldParseDatePropertyValues() throws Exception
+    {
+        // GIVEN
+        DataFactory data = data(
+                ":ID,name,date:Date\n" +
+                "0,Mattias,2018-02-27\n" +
+                "1,Johan,2018-03-01\n" );
+        Iterable<DataFactory> dataIterable = dataIterable( data );
+        Input input = new CsvInput( dataIterable, defaultFormatNodeFileHeader(), datas(),
+                defaultFormatRelationshipFileHeader(),
+                IdType.ACTUAL, config( COMMAS ), silentBadCollector( 0 ) );
+
+        // WHEN
+        try ( InputIterator nodes = input.nodes().iterator() )
+        {
+            // THEN
+            assertNextNode( nodes, 0L, new Object[]{"name", "Mattias", "date",
+                    DateValue.date( 2018, 2, 27 )}, labels() );
+            assertNextNode( nodes, 1L, new Object[]{"name", "Johan", "date",
+                    DateValue.date( 2018, 3, 1 )}, labels() );
+            assertFalse( readNext( nodes ) );
+        }
+    }
+
+    @Test
+    public void shouldParseTimePropertyValues() throws Exception
+    {
+        // GIVEN
+        DataFactory data = data(
+                ":ID,name,time:Time\n" +
+                "0,Mattias,13:37\n" +
+                "1,Johan,\"16:20:01\"\n" );
+        Iterable<DataFactory> dataIterable = dataIterable( data );
+        Input input = new CsvInput( dataIterable, defaultFormatNodeFileHeader(), datas(),
+                defaultFormatRelationshipFileHeader(),
+                IdType.ACTUAL, config( COMMAS ), silentBadCollector( 0 ) );
+
+        // WHEN
+        try ( InputIterator nodes = input.nodes().iterator() )
+        {
+            // THEN
+            assertNextNode( nodes, 0L, new Object[]{"name", "Mattias", "time",
+                    TimeValue.time( 13, 37, 0, 0, "+00:00" )}, labels() );
+            assertNextNode( nodes, 1L, new Object[]{"name", "Johan", "time",
+                    TimeValue.time( 16, 20, 1, 0, "+00:00" )}, labels() );
+            assertFalse( readNext( nodes ) );
+        }
+    }
+
+    @Test
+    public void shouldParseDateTimePropertyValues() throws Exception
+    {
+        // GIVEN
+        DataFactory data = data(
+                ":ID,name,time:DateTime\n" +
+                "0,Mattias,2018-02-27T13:37\n" +
```

Comment:
I think we need to have a case that _does_ specify an explicit timezone here, too.